win32/win32.c

```diff
@@ -622,21 +622,24 @@ init_env(void)
 
     if (!GetEnvironmentVariableW(L"HOME", env, numberof(env))) {
         f = FALSE;
-        if (GetEnvironmentVariableW(L"HOMEDRIVE", env, numberof(env)))
-            len = lstrlenW(env);
-        else
-            len = 0;
-        if (GetEnvironmentVariableW(L"HOMEPATH", env + len, numberof(env) - len) || len) {
-            f = TRUE;
-        }
-        else if (GetEnvironmentVariableW(L"USERPROFILE", env, numberof(env))) {
+        if (GetEnvironmentVariableW(L"USERPROFILE", env, numberof(env))) {
             f = TRUE;
         }
-        else if (get_special_folder(CSIDL_PROFILE, env, numberof(env))) {
-            f = TRUE;
-        }
-        else if (get_special_folder(CSIDL_PERSONAL, env, numberof(env))) {
-            f = TRUE;
+        else {
```

Comment:
Why separated only this `else`?

Reply:
The former code was made so that if either `HOMEDRIVE` or `HOMEPATH` is defined alone, then it still is used without the other half. Maybe this a questionable behavior, but I wanted to keep that how it is.

Maybe the [`diff -w`](https://github.com/ruby/ruby/pull/7033/files?diff=split&w=1) output is more descriptive.